build_docs.pl

```diff
@@ -453,6 +438,59 @@ sub check_kibana_links {
     }
 }
 
+#===================================
+sub check_elasticsearch_links {
+#===================================
+    my $build_dir    = shift;
+    my $link_checker = shift;
+    my $branch;
+    my $version;
+
+    say "Checking Elasticsearch links";
+
+    my $extractor = sub {
+        my $contents = shift;
+        return sub {
+            while ( $contents =~ m!"([^"\#]+)(?:\#([^"]+))?"!g ) {
```

Comment:
yeah. a comment is good. I could pull in a json parsing library easy enough. but then I'd have to emit this into a line per url to check and use a url parser to see it because the link checker really wants that. I think. It does feel dirty. But you got it right. Including the SHOUTING_SNAKE_CASE defense.

Reply:
Couple of questions:

- Would a different format in the ES source make this more robust? I used JSON mainly because JSON is easy in ES, but we could do something else for sure. YAML permits unquoted strings for instance. Or, have we run any Gradle stuff at this point? If so we could have a Gradle task that extracts just the bits you need into a more pleasant format.
- Does this mean we do not validate the fragment ID (the bit after the `#`)?

Reply:
One line per link in a file at the top level would amazing. But we can deal. I really could have imported a json parser, but it would have made more work in perl which I'm not familiar with. 

We are certainly *supposed* to be testing the fragment. We should be selecting it out here, but I admit to having fought with the silly regex for longer than I'd like.

Reply:
> We are certainly supposed to be testing the fragment.

Ok that's good enough for me :) I admit I gave up trying to parse the regex myself.

Reply:
Hmm I thought I'd just try this in https://github.com/elastic/elasticsearch/pull/94794 and it looks like the docs build passes even though I used an invalid fragment. It does check the bit before the `#` tho.